lib/api/initiateMultipartUpload.js

```diff
@@ -258,53 +259,73 @@ function initiateMultipartUpload(authInfo, request, log, callback) {
+                                        deletedFlag: destinationBucket.hasDeletedFlag(),
+                                    });
+                                // To avoid confusing user with error
+                                // from cleaning up
+                                // bucket return InternalError
+                                return next(errors.InternalError, corsHeaders);
+                            }
+                            return next(null, corsHeaders, destinationBucket);
                         });
-                        return callback(err, null, corsHeaders);
-                    }
-                    if (destinationBucket.hasDeletedFlag() &&
-                        accountCanonicalID !== destinationBucket.getOwner()) {
-                        log.trace('deleted flag on bucket and request ' +
-                            'from non-owner account');
-                        return callback(errors.NoSuchBucket);
-                    }
-                    if (destinationBucket.hasTransientFlag() ||
-                        destinationBucket.hasDeletedFlag()) {
-                        log.trace('transient or deleted flag so cleaning ' +
-                            'up bucket');
-                        return cleanUpBucket(destinationBucket,
-                                accountCanonicalID, log, err => {
-                                    if (err) {
-                                        log.debug('error cleaning up bucket ' +
-                                            'with flag',
-                                            { error: err,
-                                                transientFlag:
-                                            destinationBucket.hasTransientFlag(),
-                                                deletedFlag:
-                                            destinationBucket.hasDeletedFlag(),
-                                            });
-                                        // To avoid confusing user with error
-                                        // from cleaning up
-                                        // bucket return InternalError
-                                        return callback(errors.InternalError,
-                                            null, corsHeaders);
-                                    }
-                                    return _storetheMPObject(destinationBucket,
-                                        corsHeaders, objectSSEConfig);
-                                });
+            }
+            return next(null, corsHeaders, destinationBucket);
+        },
+        (corsHeaders, destinationBucket, next) =>
+            getObjectSSEConfiguration(
+                request.headers,
+                destinationBucket,
+                log,
+                (error, objectSSEConfig) => {
+                    if (error) {
+                        log.debug('error fetching server-side encryption config', {
```

Comment:
might be better to use `log.error` (unless we're sure it's already logged somewhere else in the call stack)